back/engines/commercial/content_builder/app/models/content_builder/layout_image.rb

```diff
@@ -0,0 +1,25 @@
+# == Schema Information
+#
+# Table name: content_builder_layout_images
+#
+#  id         :uuid             not null, primary key
+#  image      :string
+#  code       :string
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#
+module ContentBuilder
+  class LayoutImage < ApplicationRecord
+    mount_base64_uploader :image, LayoutImageUploader
+
+    validates :image, presence: true
+
+    before_validation :generate_code, on: :create
+
+    private
+
+    def generate_code
+      self.code ||= SecureRandom.uuid
```

Comment:
Why is `|=` required? `generate_code` is a private method, and it is invoked only `on: :create`, so `self.code` cannot be bound when invoking `generate_code`, could it?

Reply:
I would say the main reason would be to allow the creation of a layout image with a predefined code. There is no use-case for it in this PR, but if we would move to a template system for layouts, I suspect it could be desired to be able to create images with the same reference as occurs in the layout JSON.

Reply:
> if we would move to a template system for layouts

Do you have a view on if and when that would happen? I would not overdesign. If we do not need it today, don't do it. That is the [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) principle. All unused code is tech debt, so YAGNI is a simple, yet powerful, principle to make decisions.
There is no test to verify that invoking `generate_code` twice has no effect on the binding of `self.code`, so readers of the code cannot know why `||=` is here. Adding that test would explain why `||=` is here, but it would add to the tech debt pile.
So I suggest to replace `||=` with `=` until it is needed.

Reply:
Okay, I'm replacing it. One other reason I did it like this, is that we often have this kind of code throughout our models whenever we set an attribute to a default value.